### PR TITLE
Fix regression in lib/clean_number

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/1183

cc @alexcjohnson @chriddyp

### CODE REVIEW

File: src/lib/clean_number.js [hunk 1]
```diff
@@ -13,17 +13,16 @@ var isNumeric = require('fast-isnumeric');
 
 var BADNUM = require('../constants/numerical').BADNUM;
 
-// precompile these regex's for speed
-var FRONTJUNK = /^['"%,$#\s']+/;
-var ENDJUNK = /['"%,$#\s']+$/;
+// precompile for speed
+var JUNK = /['"%,$#\s]/g;
```
Comment (etpinard) [inline src/lib/clean_number.js]: Back to what it was pre #1078 on

 https://github.com/plotly/plotly.js/pull/1078/files#diff-9664b6a57e7771275ffd0c8805ea5917L30

File: test/jasmine/tests/lib_test.js [hunk 2]
```diff
@@ -1514,24 +1514,30 @@ describe('Test lib.js:', function() {
...
                 ['  $4.325  #%\t', 4.325],
                 [' " #1" ', 1],
-                [' \'\n \r -9.2e7   \t\' ', -9.2e7]
+                [' \'\n \r -9.2e7   \t\' ', -9.2e7],
+
+                // post https://github.com/plotly/plotly.js/issues/1183
+                ['2 2', 22],
+                ['2%2', 22],
+                ['2$2', 22],
+                ['1,690,000', 1690000]
```
Comment (chriddyp) [inline test/jasmine/tests/lib_test.js]: could we add one for `$3,730,000.00` to `3730000`?

Comment (chriddyp) [inline test/jasmine/tests/lib_test.js]: Also: `" $1,410,000.00 "` to `3,870,000.00`?

Comment (etpinard) [inline test/jasmine/tests/lib_test.js]: What about `'1 000 000'` ?

**Comment by alexcjohnson** [comment]
Since we're reverting this to buy us some time... lets do this right. Can we do:
`var JUNK = /(^['"%,$#\s]|,|['"%,$#\s]/g`
ie strip only commas in the middle, and all these other characters on the ends?

**Comment by etpinard** [comment]
Closed in favour of https://github.com/plotly/plotly.js/pull/1185

**Pull request closed without merging**